### PR TITLE
:lipstick: :bug: Improve auth modal UX and fix error handling

### PR DESCRIPTION
**Pull request by locness3**
Peek 2021-02-10 18-48
- Mark the inputs as "required"
- Add a loadin spinner to the button
- Disable the button if the two inputs aren't filled out
- Actually handle HTTP errors: before, it was trying to parse JSON no matter what - only parse JSON if HTTP response is 200

### CODE REVIEW

**Comment by locness3** [comment]
I made this mostly because the login is pretty slow - when I tested it out it looked like it didn't do anything.

/cc @micahlt

**Comment by micahlt** [comment]
@locness3 Incredible!  Thanks for the PR 😄 A few things - first, can we decrease the size of the spinner when the sign-in button is clicked?  It would be nice for it to be a little larger than the lock icon is in order to prevent a large layout shift.  Otherwise, this looks ready-to-merge!  Also, make sure that you update the semver number in `src/views/Tab4.vue` and both the `versionName` and `versionCode` in `android/app/build.gradle`.  The `versionCode` should just increase by one.  Thanks!

**Comment by locness3** [comment]
> can we decrease the size of the spinner

>  It would be nice for it to be a little larger is in order to prevent a large layout shift

... So, do we decrease or increase it? What would prevent a layout shift, not increasing it too much?

**Comment by locness3** [comment]
I plan on doing more stuff (mostly refinements), so I'll bump the semver to `0.5.0`.

**Comment by micahlt** [comment]
Sorry, I'm being confusing!  I mean we should _shrink_ the loader to where it's only a slight bit larger than the lock (smaller than it is now).

**Comment by locness3** [comment]
Okay! ~~Do it~~

File: src/components/AuthModal.vue [hunk 1]
```diff
@@ -20,17 +20,19 @@
...
+          <ion-input type="password" required v-model="password"></ion-input>
         </ion-item>
         <ion-grid>
           <ion-row>
             <ion-col style="text-align:center;">
-              <ion-button size="medium" expand="block" @click="logIn">
-                <ion-icon :icon="lockClosed" style="font-size:15px;padding-right:10px;"></ion-icon>Sign In
+              <ion-button size="medium" expand="block" :disabled="!username || !password" @click="logIn">
+                <ion-icon v-if="!loading" :icon="lockClosed" style="font-size:15px;padding-right:10px;"></ion-icon>
+                <ion-spinner v-if="loading" style="margin-right:10px"></ion-spinner>
```
Comment (micahlt) [inline src/components/AuthModal.vue:34]: Looks like we need this line to be changed to shrink the spinner - maybe an `em` measurement for the height and width might work well?

Comment (micahlt) [reply src/components/AuthModal.vue:34]: @locness3

Comment (locness3) [reply src/components/AuthModal.vue:34]: I'll try `em`

Comment (locness3) [reply src/components/AuthModal.vue:34]: image
Like this? @micahlt

**Comment by micahlt** [comment]
Looks great!  Merging now.  Also, get ready for a big feature 🤭

**Comment by locness3** [comment]
Should've changed the text while I was at it

**Pull request merged**